src/tests/unit/enterprise_edr/test_enterprise_edr_events.py

```diff
@@ -87,3 +87,22 @@ def test_event_query_select_with_where(cbcsdk_mock):
     assert len(results) == 100
     first_result = results[0]
     assert first_result.process_guid == guid
+
+
+def test_event_query_select_asynchronous(cbcsdk_mock):
+    """Test Event Querying with where() clause as asynchronous"""
+    search_validate_url = "/api/investigate/v1/orgs/test/events/search_validation"
+    cbcsdk_mock.mock_request("GET", search_validate_url, EVENT_SEARCH_VALIDATION_RESP)
+
+    url = r"/api/investigate/v2/orgs/test/events/J7G6DTLN\\-006633e3\\-00000334\\-00000000\\-1d677bedfbb1c2e/_search"
+    cbcsdk_mock.mock_request("POST", url, EVENT_SEARCH_RESP)
+
+    api = cbcsdk_mock.api
+    guid = "J7G6DTLN-006633e3-00000334-00000000-1d677bedfbb1c2e"
+
+    events = api.select(Event).where(process_guid=guid)
+    future = events.execute_async()
+    results = future.result()
+    assert len(results) == 250  # we get all of them when we run in background
```

Comment:
This is only partially true since events works slightly different that the other search APIs. We need to check if `processed_segments == total_segements` 

Here is a link to the EA which required us to expose this property so they could retry the request to make sure they get the entire results back https://jira.carbonblack.local/browse/EA-16415?focusedCommentId=1230189&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-1230189